### PR TITLE
Move badge styling into ha-badge component to reuse it

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Extract badge style into `ha-badge` component to reuse it.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

## Summary by CodeRabbit

- **New Features**
	- Introduced the `HaBadge` component for creating customizable badges and buttons.
	- Enhanced error handling by integrating the `HaBadge` component into existing badge structures.

- **Bug Fixes**
	- Improved rendering logic for badges, ensuring better error display and interaction.

- **Refactor**
	- Replaced previous badge implementations with the new `HaBadge` component for better modularity and maintainability.
	- Streamlined the internal structure of badges to enhance code readability and functionality.

- **Style**
	- Updated styles to target the new `HaBadge` component, ensuring consistent visual behavior across badges.

### CODE REVIEW

File: src/components/ha-badge.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
+import { css, CSSResultGroup, html, LitElement, nothing } from "lit";
+import { customElement, property } from "lit/decorators";
+import { classMap } from "lit/directives/class-map";
+import { ifDefined } from "lit/directives/if-defined";
+import "./ha-ripple";
+
+type BadgeType = "badge" | "button";
+
+@customElement("ha-badge")
```
Comment (balloob) [inline src/components/ha-badge.ts:9]: So now we'll have `hui-badge` and `ha-badge` 😬 Feels a bit chaotic. 

Should we rename `hui-badge` to something that represents it's creating one from config?

File: src/components/ha-badge.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        role=${ifDefined(this.type === "button" ? "button" : undefined)}
+        tabindex=${ifDefined(this.type === "button" ? "0" : undefined)}
+      >
+        <ha-ripple .disabled=${this.type !== "button"}></ha-ripple>
+        <slot name="icon"></slot>
+        ${this.iconOnly
+          ? nothing
+          : html`<span class="info">
+              ${label ? html`<span class="label">${label}</span>` : nothing}
+              <span class="content"> <slot></slot></span>
```
Comment (balloob) [inline src/components/ha-badge.ts:34]:
```suggestion
              <span class="content"><slot></slot></span>
```

**Review by balloob — DISMISSED** [review]
Nice! 👍

File: src/components/ha-badge.ts (hunk 1 above)
Comment (piitaya) [reply src/components/ha-badge.ts:9]: We created `hui-card` and `hui-badge` because we had `hui-view` 😅. I agree, it's super confusing.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**